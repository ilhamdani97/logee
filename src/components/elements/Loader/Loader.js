import React from 'react';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import compareProps from '../../../utils/compareProps';

function Loader(props) {
  const root = [props.className, styles.root].join(' ');
  return (
    <div {...props} className={root}>
      <div className={styles.loading}/>
    </div>
  );
}

export default React.memo(Loader, compareProps);

Loader.defaultProps = {
  className: ''
};

Loader.propTypes = {
  className: PropTypes.string
};
