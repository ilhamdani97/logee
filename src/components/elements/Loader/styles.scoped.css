.root {
  height: 20rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.root > .loading {
  border: .5rem solid #f3f3f3;
  border-top: .5rem solid var(--primary-color);
  border-radius: 50%;
  width: 3rem;
  height: 3rem;
  animation: spin 2s linear infinite;
}