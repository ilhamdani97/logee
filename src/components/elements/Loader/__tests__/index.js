import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Loader from '../Loader';

const { defaultProps } = Loader.type;

describe('src/components/elements/Loader', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Loader {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
