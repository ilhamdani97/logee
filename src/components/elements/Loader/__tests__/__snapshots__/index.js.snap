// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Loader render 1`] = `
<div
  className=" "
>
  <div />
</div>
`;
